File: lib/core/Random.py
#!/usr/bin/env python

import random


class RandomGenerator:
    @staticmethod
    def RandomString(Length=10):
        return "".join(
            random.choice("abcdefghijklmnopqrstuvwxyz0123456789") for _ in range(Length)
        )

    @staticmethod
    def RandomIP():
        return f"{random.randint(1, 254)}.{random.randint(1, 254)}.{random.randint(1, 254)}.{random.randint(1, 254)}"

    @staticmethod
    def CalculateChecksum(Data):
        Sum = 0
        for I in range(0, len(Data), 2):
            if I + 1 < len(Data):
                Sum += (Data[I] << 8) + Data[I + 1]
            else:
                Sum += Data[I] << 8
        Sum = (Sum >> 16) + (Sum & 0xFFFF)
        Sum += Sum >> 16
        return ~Sum & 0xFFFF

File: internal/random/random.go
package random

import (
	"encoding/binary"
	"fmt"
	"math/rand"
)

const Charset = "abcdefghijklmnopqrstuvwxyz0123456789"

func RandomString(Length int) string {
	B := make([]byte, Length)
	for I := range B {
		B[I] = Charset[rand.Intn(len(Charset))]
	}
	return string(B)
}

func RandomIP() string {
	return fmt.Sprintf("%d.%d.%d.%d",
		rand.Intn(254)+1,
		rand.Intn(254)+1,
		rand.Intn(254)+1,
		rand.Intn(254)+1,
	)
}

func RandomInt(Min, Max int) int {
	return Min + rand.Intn(Max-Min+1)
}

func RandomBytes(N int) []byte {
	B := make([]byte, N)
	rand.Read(B)
	return B
}

func CalculateChecksum(Data []byte) uint16 {
	Sum := 0
	for I := 0; I+1 < len(Data); I += 2 {
		Sum += int(binary.BigEndian.Uint16(Data[I : I+2]))
	}
	if len(Data)%2 != 0 {
		Sum += int(Data[len(Data)-1]) << 8
	}
	for Sum>>16 != 0 {
		Sum = (Sum & 0xffff) + (Sum >> 16)
	}
	return ^uint16(Sum)
}
